### PR TITLE
Filtering by event types

### PR DESCRIPTION
**Pull request by mpraglowski**
It solves #456

### CODE REVIEW

**Comment by mpraglowski** [comment]
@joelvh again need your help here :( Memory adapter needs some attention.

**Comment by joelvh** [comment]
Fixed in #475

**Comment by joelvh** [comment]
@mpraglowski were you able to merge my changes in?

**Comment by mpraglowski** [comment]
@joelvh yes - I will close #475 - your changes are already here

**Comment by joelvh** [comment]
Awesome, thanks!

**Comment by pawelpacana** [comment]
Worth reminding: https://github.com/RailsEventStore/rails_event_store/issues/72

File: ruby_event_store-rom/db/migrate/20181026152045_index_by_event_type.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+require 'rom/sql'
+
+::ROM::SQL.migration do
+  change do
+    alter_table(:event_store_events) do
+      add_index :event_type
```
[the remark this answers is not in the document]
Comment (joelvh) [reply ruby_event_store-rom/db/migrate/20181026152045_index_by_event_type.rb]: ROM uses Sequel under the hood. You could use `add_index :event_type, name: 'event_store_events_event_type_index', if_not_exists : true`

The  `if_not_exists` option is PG only. Would that work? (Seems MySQL requires a query or stored procedure to check if the index exists by name.)

Also, maybe use the index name that Rails generates in the AR migration?

https://sequel.jeremyevans.net/rdoc/classes/Sequel/Schema/AlterTableGenerator.html#method-i-add_index

File: rails_event_store_active_record-legacy/lib/rails_event_store_active_record/legacy/generators/templates/v1_v2_migration_template.rb [hunk 2]
```diff
@@ -71,6 +70,7 @@ def up
       end
       add_index :event_store_events, :id, unique: true if rails_42
       add_index :event_store_events, :created_at
+      add_index :event_store_events, :event_type
```
Comment (pawelpacana) [inline rails_event_store_active_record-legacy/lib/rails_event_store_active_record/legacy/generators/templates/v1_v2_migration_template.rb]: Isn't it redundant? We don't remove existing index from legacy schema and add a new one on the same column?

Comment (pawelpacana) [reply rails_event_store_active_record-legacy/lib/rails_event_store_active_record/legacy/generators/templates/v1_v2_migration_template.rb]: Oh, sqlite...

**Pull request merged**